src/item.cpp

```diff
@@ -5421,6 +5422,13 @@ bool item::process_litcig( player *carrier, const tripoint &pos )
             g->m.add_item_or_charges( tripoint( pos.x + rng( -1, 1 ), pos.y + rng( -1, 1 ), pos.z ), *this, 2 );
             return true; // removes the item that has just been added to the map
         }
+
+        if(  carrier->has_effect( effect_sleep ) ) {
+            carrier->add_msg_if_player( m_bad, _( "You fall asleep and drop your %s." ),
+                                        tname().c_str() );
+            g->m.add_item_or_charges( tripoint( pos.x + rng( -1, 1 ), pos.y + rng( -1, 1 ), pos.z ), *this, 2 );
```

Comment:
Just a guess about potential issue: what happens when this tripoint refers on tile which cannot contain any items?